frontend/src/components/userPage/UserPageContent.jsx

```diff
@@ -23,11 +23,24 @@ const UserPageContent = () => {
     const [date, setDate] = useState('')
     const [time, setTime] = useState('')
     // ---------------------------------------
+    const [userPosition, setUserPosition] = useState({
+        latitude: 0,
+        longitude: 0,
+    })
+    let [customLocation, setCustomLocation] = useState(false)
 
     useEffect(() => {
         if (changeInEvent) {
             fetchUserEvents()
         }
+        if ('geolocation' in navigator && !customLocation) {
+            navigator.geolocation.getCurrentPosition(function (position) {
+                setUserPosition({
+                    latitude: position.coords.latitude,
+                    longitude: position.coords.longitude,
+                })
+            })
+        }
```

Comment:
So user location is updated whenever userEvents state is updated? Can you elaborate on this logic?

Reply:
To do:
Fix useEffect UserPageContent.jsx

Reply:
Done

Reply:
> So user location is updated whenever userEvents state is updated? Can you elaborate on this logic?

I did this because when I used console.log(userEvents) the first 3 or 4 times it appeared in blank, and I was having errors when loading the page for the first time. Fetching it like this solved the problem.
**Resolved in PR 35**